expression/builtin.go

```diff
@@ -362,6 +412,21 @@ func (b *baseStringBuiltinFunc) evalDuration(row []types.Datum) (types.Duration,
 	panic("cannot get DURATION result from ClassString expression")
 }
 
+func (b *baseStringBuiltinFunc) getRetTp() *types.FieldType {
+	tp, flen := b.tp, b.tp.Flen
+	if flen >= mysql.MaxBlobWidth {
+		tp.Tp = mysql.TypeLongBlob
+	} else if flen >= 65536 {
```

Comment:
I can't find the real meaning of this number in [MySQL source code](https://github.com/mysql/mysql-server/blob/5.7/sql/item.cc#L6314),
so I didn't convert it to constant.